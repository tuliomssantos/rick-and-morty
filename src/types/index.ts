export type FetchCharactersInput = {
  page: number
  filter?: {
    name?: string
    status?: string
    species?: string
    type?: string
    gender?: string
  }
}

export type CharacterPreviewData = {
  id: string
  image: string
  location: {
    name: string
  }
  name: string
  status: string
  species: string
}

export type CharactersData = {
  characters: {
    info: {
      count: number
      pages: number
    }
    results: Array<CharacterPreviewData>
  }
}

export type ColorVariant =
  | 'primary'
  | 'secondary'
  | 'error'
  | 'warning'
  | 'info'
  | 'success'

export type ColorSubvariant = 'dark' | 'light'
