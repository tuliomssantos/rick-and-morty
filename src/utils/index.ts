export * from './createEmotionCache'
export * from './renderWithTeme'
