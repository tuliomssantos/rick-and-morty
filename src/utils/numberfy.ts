export interface INumberfy {
  (value: string | string[] | undefined, defaultValue: number): number
}

export const numberfy: INumberfy = (value, defaultValue) => {
  if (value) {
    return Number(value)
  }

  return defaultValue
}
