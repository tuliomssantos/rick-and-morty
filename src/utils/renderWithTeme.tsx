import { render, RenderResult } from '@testing-library/react'

import { CustomThemeProvider } from '../styles/theme'

export const renderWithTheme = (children: React.ReactNode): RenderResult =>
  render(<CustomThemeProvider>{children}</CustomThemeProvider>)
