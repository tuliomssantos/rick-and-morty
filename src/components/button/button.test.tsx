import Button from './'

import { renderWithTheme } from '@/src/utils'

it('renders Button unchanged', () => {
  const { container } = renderWithTheme(<Button />)
  expect(container.firstChild).toMatchSnapshot()
})
