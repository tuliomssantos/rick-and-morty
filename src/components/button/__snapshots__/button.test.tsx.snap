// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Button unchanged 1`] = `
<button
  class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-s1n7pn-MuiButtonBase-root-MuiButton-root"
  tabindex="0"
  type="button"
>
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;
